### PR TITLE
Reenable forked mainnet

### PR DESCRIPTION
**Pull request by peterargue**
Closes #???

## Description

This PR reenables forked network support and updates it to use Access nodes as the archive backend

______

For contributor use:

- [ ] Targeted PR against `master` branch
- [x] Linked to GitHub issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/flow-emulator/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation
- [x] Re-reviewed `Files changed` in the GitHub PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -190,12 +199,14 @@ require (
...
 	lukechampine.com/blake3 v1.2.1 // indirect
 	modernc.org/libc v1.22.3 // indirect
 	modernc.org/mathutil v1.5.0 // indirect
 	modernc.org/memory v1.5.0 // indirect
 	modernc.org/sqlite v1.21.1 // indirect
 	nhooyr.io/websocket v1.8.7 // indirect
+	rsc.io/tmplfunc v0.0.3 // indirect
 )
+
+// replace github.com/onflow/flow-go => ../flow-go
```
Comment (turbolent) [inline go.mod:212]:
```suggestion
```

**Review by turbolent — COMMENTED** [review]
Thank you for bringing this back! 🙏 

There are still a couple commented out lines (debug statements, error handling) that might need to be addressed

**Comment by turbolent** [comment]
Once merged, can you please also port this to `master`?

**Comment by turbolent** [comment]
Re: target branch: Could you please sync with #566 and #564 so all PRs go into the same "pre-C1.0" branch and we do not end up with multiple different ones?

File: storage/remote/store.go [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+}
+
+type Option func(*Store)
+
+// WithChainID sets a chain ID and is used to determine which archive node to use.
+func WithChainID(ID flowgo.ChainID) Option {
+	return func(store *Store) {
+		archiveHosts := map[flowgo.ChainID]string{
+			flowgo.Mainnet: "access-008.mainnet24.nodes.onflow.org:9000",
+			flowgo.Testnet: "access-002.devnet49.nodes.onflow.org:9000",
```
Comment (bluesign) [inline storage/remote/store.go:59]: should we update those?

File: go.mod (hunk 1 above)
Comment (peterargue) [inline go.mod:212]:
```suggestion
```

**Review by bluesign — APPROVED** [review]

**Comment by turbolent** [comment]
@peterargue There are several such "pre-Cadence 1.0" branches now, e.g. see https://github.com/onflow/flow-emulator/pull/566#issuecomment-1937801361 and https://github.com/onflow/flow-emulator/pull/564#issuecomment-1937863473.

I'm not saying that `pre-cadence-1.0` isn't it, all I'm trying to avoid is that there are several such branches that get out of sync.

**Comment by tarakby** [comment]
I would need to use a pre-cadence-1.0 branch and I was thinking of using this one: https://github.com/onflow/flow-emulator/tree/tarak/v0.58.1-0-with-crypto-type-fixes (that's the one used by flow-go master)

However, if these two serve the same purpose, can we consolidate them into one? @peterargue

**Comment by UlyanaAndrukhiv** [comment]
@peterargue, @tarakby,  @chasefleming, I want to sync with you about what the appropriate target branch we should use for continuing releases before reaching version 1.0.0. 
I need to extend the flow-emulator according to my `flow-go` PR (https://github.com/onflow/flow-go/pull/5307), and I find myself uncertain which target branch should use:  `release/v0 `,  `tarak/v0.58.1-0-with-crypto-type-fixes` or `pre-cadence-1.0`?  
I have observed that the `release/v0` branch lacks the necessary upgrades related to the crypto package, which are already incorporated into the `tarak/v0.58.1-0-with-crypto-type-fixes` branch.

**Comment by peterargue** [comment]
I recreated a new PR https://github.com/onflow/flow-emulator/pull/594 that targets the `release/v0` branch

**Pull request closed without merging**